modules/csg/csg_shape.cpp

```diff
@@ -492,6 +494,50 @@ PoolVector<Vector3> CSGShape::get_brush_faces() {
 	return faces;
 }
 
+bool CSGShape::split_by_surface(Vector<Variant> p_destination_mesh_instances) {
+	ERR_FAIL_COND_V_MSG(!is_inside_tree(), false, "Source CSGShape must be inside the SceneTree.");
+
+	// For simplicity we are requiring that the destination MeshInstances have the same parent
+	// as the source. This means we can use identical transforms.
+	Node *parent = get_parent();
+	ERR_FAIL_COND_V_MSG(!parent, false, "Source CSGShape must have a parent node.");
+
+	// bound function only support variants, so we need to convert to a list of MeshInstances
+	Vector<MeshInstance *> mis;
+
+	for (int n = 0; n < p_destination_mesh_instances.size(); n++) {
+		MeshInstance *mi = Object::cast_to<MeshInstance>(p_destination_mesh_instances[n]);
+		if (mi) {
+			ERR_FAIL_COND_V_MSG(mi->get_parent() != parent, false, "Destination MeshInstances must be siblings of the source CSGShape.");
+			mis.push_back(mi);
+		} else {
+			ERR_FAIL_V_MSG(false, "Only MeshInstances can be split.");
+		}
+	}
+
+	force_update_shape();
+
+	Array arr = get_meshes();
+	ERR_FAIL_COND_V(!arr.size(), false);
```

Comment:
```suggestion
	ERR_FAIL_COND_V(arr.is_empty(), false);
```
Would generally advocate in favour of this, what do you think?

Reply:
Am happy to change. :+1: 

I personally don't have strong views either way - it's syntactic sugar I think, and probably compiles down to the same (or similar), but some may find it easier to read.